sahi/predict.py

```diff
@@ -416,6 +424,10 @@ def predict(
 
     # for profiling
     durations_in_seconds = dict()
+    
+    # Need this line, if we don't give images to model 
+    if source is None:
```

Comment:
Can you elaborate more why we need this, isn't [this](https://github.com/obss/sahi/blob/b7fb59d6a41260d8a6318549efcb4d4cb9718bf5/sahi/predict.py#L431) sufficient ?

Reply:
Normally, `source` is used for image inference and this is many parts of code. If we give input video, `source` variable is none. The code give error to us at this situation, so I used it like that. But If we inference on image, source is not none and this not problem. Have you better idea for this? Because my idea is quick solution.

Reply:
@madenburak can we make it like this?
1. perform image inference using `source` when `export_video=False`
2. perform video inference using `source` when `export_video=True`

Reply:
Actually, the method that you say is more useful. I will edit the code with this way when I have time.